### PR TITLE
Improve GitHub templates and VSCode config

### PR DESCRIPTION
**Pull request by StephanTLavavej**
# Description

* Move issue templates to `.github` and update them.
  * In the Bug Report template, the "STL version" section was easy to miss. This moves it down (as it's usually not critical info) and expands it so it's easy to understand and hard to miss.
  * Add a cxx20 Feature template, including an explanation of "broken links". This is an alternative to #582. Thanks to @SuperWig for the suggestion!
* Move `CODEOWNERS` and `pull_request_template.md` to `.github`.
* Slightly improve `pull_request_template.md`. During editing, GitHub wants to suggest issue/PR numbers when the cursor is over `#`. Using `===` syntax for headers avoids this.
* VSCode: Set `C_Cpp.autoAddFileAssociations` to `false`. This "Controls whether files are automatically added to `files.associations` when they are the target of a navigation operation from a C/C++ file."

# Checklist

Be sure you've read README.md and understand the scope of this repo.

If you're unsure about a box, leave it unchecked. A maintainer will help you.

- [x] Identifiers in product code changes are properly `_Ugly` as per
  https://eel.is/c++draft/lex.name#3.1 or there are no product code changes.
- [x] The STL builds successfully and all tests have passed (must be manually
  verified by an STL maintainer before automated testing is enabled on GitHub,
  leave this unchecked for initial submission).
- [x] These changes introduce no known ABI breaks (adding members, renaming
  members, adding virtual functions, changing whether a type is an aggregate
  or trivially copyable, etc.).
- [x] These changes were written from scratch using only this repository,
  the C++ Working Draft (including any cited standards), other WG21 papers
  (excluding reference implementations outside of proposed standard wording),
  and LWG issues as reference material. If they were derived from a project
  that's already listed in NOTICE.txt, that's fine, but please mention it.
  If they were derived from any other project (including Boost and libc++,
  which are not yet listed in NOTICE.txt), you *must* mention it here,
  so we can determine whether the license is compatible and what else needs
  to be done.

### CODE REVIEW

**Comment by StephanTLavavej** [comment]
If you want to see what the new issue templates look like in practice, look at my fork https://github.com/StephanTLavavej/STL where I have committed these changes to the default branch.

File: .github/ISSUE_TEMPLATE/cxx20-feature.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+Please use the Bug Report issue template instead.
+
+
+[PAPER_NUMBER](https://wg21.link/PAPER_NUMBER) PAPER_TITLE
+
+Note: If this link appears to be broken, don't worry! Papers are often
+revised during a WG21 meeting before the final vote. When that happens,
+there's a delay of a few weeks between the end of the meeting
+and the publication of the papers. When the papers are available,
+the https://wg21.link redirector will start working.
```
Comment (CaseyCarter) [inline .github/ISSUE_TEMPLATE/cxx20-feature.md]: I've noticed that wg21.link redirects to isocpp.org for papers submitted using the new automated system even before the mailing goes out (e.g., https://wg21.link/p2091r0). I don't know that this fact merits mention here, but FYI.

Comment (StephanTLavavej) [reply .github/ISSUE_TEMPLATE/cxx20-feature.md]: Yep, I've noticed that too. I think we don't need to mention the complexity - the note begins with "If this link appears to be broken" and nobody's going to complain if it works.

**Comment by StephanTLavavej** [comment]
FYI, I've pushed a microscopic change after approval, to mark `.clang-format` as YAML. I noticed that I had done this in our MS-internal VSCode settings, but not on GitHub.

**Comment by StephanTLavavej** [comment]
@SuperWig, thanks again for thinking about ways to avoid reader confusion! 😺

**Pull request merged**